pkg/logs/scheduler/scheduler.go

```diff
@@ -206,6 +206,8 @@ func (s *Scheduler) toSources(config integration.Config) ([]*logsConfig.LogSourc
 	}
 
 	configName := s.configName(config)
+	// Used to fill empty source/service in some cases
+	serviceName, sourceName := s.extractMainSourceAndService(configs)
```

Comment:
nit: Would this be better to put in the specific block where it's used? It seems like it's only useful for a small slice of configurations so having it out here seem like it's associating it with a much larger context without a need.

Reply:
We need to run it against the whole `configs` slice, note that the `configs` slice will be rather small (like < 5) most of the time (it comes from a single annotation/label). I tried to make it more explicit.